### ISSUE TITLE
Does gradient accumulation affect prodigy?

### ISSUE THREAD

**Issue by lyndonlauder** [LoganBooker/prodigy-plus-schedule-free#2]
Hi, i would like to try prodigy and schedulefree prodigy. I gradually increase the batch size during training with gradient accumulation, do you know if this will affect prodigy in any negative way since I see in your repo description "in terms of finding a good LR, which it usually achieves after ~25% of training, though this is very dependent on batch size and epochs."

I know gradient can sometimes not work properly when some batch specific code/calculation. Similar to how batch normalization only works on mini batches and not accumulated batches.

Thanks for your efforts!

**Comment by LoganBooker** [comment]
Hi @lyndonlauder, thanks for your interest!

Upping the gradient accumulation to increase your effective batch size should have identical behaviour to upping the actual batch size, so in terms of training dynamics, it should work out the same. In terms of any specific behaviour for Prodigy and ScheduleFree, it doesn't have any built-in numbers that are based on batch size.

So, long story short, you should be fine! My recommendation is for each new dataset/batch size/rank-alpha, is always to monitor the value of `d` for 1-2 training runs and see where it "plateaus" (in my experience, this is usually within the first ~25% of training as I mentioned). Most of the time, you can switch Prodigy off after that (via the `prodigy_steps` setting).
